flink-runtime/src/main/java/org/apache/flink/streaming/api/operators/InternalTimerServiceSerializationProxy.java

```diff
@@ -128,6 +128,6 @@ private <N> InternalTimerServiceImpl<K, N> registerOrGetTimerService(
                         restoredTimersSnapshot.getNamespaceSerializerSnapshot().restoreSerializer();
         TimerSerializer<K, N> timerSerializer =
                 new TimerSerializer<>(keySerializer, namespaceSerializer);
-        return timerServicesManager.registerOrGetTimerService(serviceName, timerSerializer);
+        return timerServicesManager.restoreTimeService(serviceName, timerSerializer);
```

Comment:
How about keeping the original name?

Reply:
Removed `restoreTimeService`, and move the check of `priorityQueueSetFactory` into `registerOrGetTimerService`.